src/index.ts

```diff
@@ -119,7 +131,10 @@ async function initVueRouterGuard(
       }
     }
 
-    const additionalEventData: Record<string, any> = (to.meta?.gtmAdditionalEventData as Record<string, any>) ?? {};
+    const additionalEventData: Record<string, any> = {
+      ...deriveAdditionalEventData(to, from),
```

Comment:
We are already in an async context here. I think we can improve this with letting `deriveAdditionalEventData` be an async function and await it here.

We need to typedef `vueRouterAdditionalEventData` as `?: (to: RouteLocationNormalized, from: RouteLocationNormalized) => Record<string, any> | Promise<(to: RouteLocationNormalized, from: RouteLocationNormalized) => Record<string, any>>;` for this. Or maybe we should extract the callback type to a type and then reuse it.

Hope you know what I mean :D

Reply:
This happens within the callback to `vueRouter.afterEach` which is currently not `async`. Only the outer function is at the moment. We can make the callback to `vueRouter.afterEach` asynchronous though, I think that should work. What do you think?

Reply:
I think thios would be a nice to have / cool addition :slightly_smiling_face:

Reply:
Added the suggestion

Reply:
nice :raised_hands: